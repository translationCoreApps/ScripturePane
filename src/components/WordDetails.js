import React from 'react';
import PropTypes from 'prop-types';
// helpers
import * as lexiconHelpers from '../helpers/lexiconHelpers';

class WordDetails extends React.Component {

  render() {
    let {lemma, morph, strongs} = this.props.word;

    let lexicon;
    const {lexicons} = this.props.resourcesReducer;
    const entryId = lexiconHelpers.lexiconEntryIdFromStrongs(strongs);
    const lexiconId = lexiconHelpers.lexiconIdFromStrongs(strongs);
    if (lexicons[lexiconId] && lexicons[lexiconId][entryId]) {
      lexicon = lexicons[lexiconId][entryId].long;
    }
    return (
      <div style={{margin: '-10px 10px -20px', maxWidth: '400px'}}>
        <span><strong>Lemma:</strong> {lemma}</span><br/>
        <span><strong>Morphology:</strong> {morph}</span><br/>
        <span><strong>Strongs:</strong> {strongs}</span><br/>
        <span><strong>Lexicon:</strong> {lexicon}</span><br/>
      </div>
    );
  }
}

WordDetails.propTypes = {
  word: PropTypes.object.isRequired,
  resourcesReducer: PropTypes.object.isRequired
};

export default WordDetails;
