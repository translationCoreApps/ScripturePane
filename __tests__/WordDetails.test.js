import React from 'react';
import renderer from 'react-test-renderer';
// components
import WordDetails from '../src/components/WordDetails';

describe('WordDetails component', () => {
  test('renders correctly', () => {
    const props = {
      word: {
        lemma: "καί",
        morph: "Gr,CC,,,,,,,,",
        strong: "G25320"
      },
      lexiconData: {
        'ugl': {
          '2532': {
            "brief": "and, even, also, namely",
            "long": "and, even, also, namely."
          }
        }
      }
    };
    const tree = renderer
      .create(<WordDetails {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
