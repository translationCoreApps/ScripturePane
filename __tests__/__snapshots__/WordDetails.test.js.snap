// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordDetails component renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "-10px 10px -20px",
      "maxWidth": "400px",
    }
  }
>
  <span>
    <strong>
      Lemma:
    </strong>
     
    καί
  </span>
  <br />
  <span>
    <strong>
      Morphology:
    </strong>
     
    Gr,CC,,,,,,,,
  </span>
  <br />
  <span>
    <strong>
      Strongs:
    </strong>
     
    G25320
  </span>
  <br />
  <span>
    <strong>
      Lexicon:
    </strong>
     
    and, even, also, namely.
  </span>
  <br />
</div>
`;
